bundler/lib/bundler/plugin.rb

```diff
@@ -100,23 +102,28 @@ def list
     #
     # @param [Pathname] gemfile path
     # @param [Proc] block that can be evaluated for (inline) Gemfile
-    def gemfile_install(gemfile = nil, &inline)
-      Bundler.settings.temporary(frozen: false, deployment: false) do
+    def gemfile_install(gemfile = nil, lockfile = nil, unlock = {}, &inline)
+      # skip the update if unlocking specific gems, but none of them are our plugins
+      if unlock.is_a?(Hash) && unlock[:gems] && !unlock[:gems].empty? &&
+         (unlock[:gems] & index.installed_plugins).empty?
+        unlock = {}
+      end
+
+      Bundler.settings.temporary(plugin_gemfile_parse: true, plugins_in_lockfile: true) do
```

Comment:
For `plugin_gemfile_parse`, I can certainly use a class instance variable on Bundler::Plugin. I hesitated to do that because that's essentially a global variable, but I guess so is Bundler.settings.temporary.

For `plugins_in_lockfile`, I'm happy to change the name, but keep in mind this is the same as the user-facing setting to enable the new behavior. During this phase, we _always_ have to have that setting enabled otherwise no plugins would get installed, ever!

Reply:
> For plugin_gemfile_parse, I can certainly use a class instance variable on Bundler::Plugin. I hesitated to do that because that's essentially a global variable, but I guess so is Bundler.settings.temporary.

Yeah, that's my thinking too, I would use a class instance variable to avoid exposing a setting.

> Regarding `plugins_in_lockfile: true`, it makes me think if it's a good name. My understanding is that this phase does not use a lockfile at all, so maybe resolve_plugins_as_regular_gems or something like that would fit better?

Yep, I'm aware. I was thinking of finding a name that's good for end users but also unsurprising when reading the source code here. It surprised me that we need to unconditionally enable "plugins in lockfile" in this phase, when there's no lockfile. I don't have a very strong opinion though.

Reply:
I'll do the class instance variable, and leave the other setting the same, but add an explanatory comment about why it needs to be temporarily set here.